addons/l10n_es_edi_facturae/models/account_move.py

```diff
@@ -192,7 +192,10 @@ def _l10n_es_edi_facturae_inv_lines_to_items(self, conversion_rate=None):
                 continue
             invoice_line_values = {}
 
-            price_before_discount = line.currency_id.round(line.price_subtotal / (1 - line.discount / 100.0))
+            if line.discount == 100:
+                price_before_discount = line.currency_id.round(line.quantity * line.price_unit)
+            else:
+                price_before_discount = line.currency_id.round(line.price_subtotal / (1 - line.discount / 100.0))
```

Comment:
Is this actually correct to go backward like this? Shouldn't we use the `price_unit` everywhere in the first place?

Reply:
@JulienAlardot

Reply:
The price_unit is (possibly or not) tax included, this one isn't so no. Or we would go even more backward

Reply:
but that's what you take in the 100% discount scenario so does it means this line is not correct for taxes? I feel like trying to go reverse will trigger rounding issues (e.g. on a 99% discount on a small amount).

Reply:
Yeah, I suppose something like this could work (even though it's basically computing the tax twice) (not tested)

```suggestion
                tax_base_before_discount = self.env['account.tax']._convert_to_tax_base_line_dict(
                    currency=line.currency_id,
                    taxes=line.tax_ids,
                    price_unit=line.price_unit,
                    quantity=line.quantity,
                )
                tax_before_discount = self.env['account.tax']._compute_taxes([tax_base_before_discount])
                price_before_discount = sum(to_update['price_subtotal'] for _dummy, to_update in tax_before_discount['base_lines_to_update'])

Reply:
@JulienAlardot Changes are done!!